deepinv/datasets/datagenerator.py

```diff
@@ -58,69 +95,168 @@ def __init__(
         dtype: torch.dtype = torch.float,
         complex_dtype: torch.dtype = torch.cfloat,
     ):
+        import h5py
+
         super().__init__()
-        self.data_info = []
-        self.data_cache = {}
-        self.unsupervised = False
+
+        f = h5py.File(path, "r")
+
+        self._register_members(
+            f,
+            train=train,
+            split=split,
+            load_physics_generator_params=load_physics_generator_params,
+        )
+
+        self.hd5 = f
         self.transform = transform
-        self.load_physics_generator_params = load_physics_generator_params
         self.cast = lambda x: x.type(complex_dtype if x.is_complex() else dtype)
 
-        if isinstance(h5py, ImportError):
-            raise h5py
-
-        self.hd5 = h5py.File(path, "r")
-        suffix = ("_train" if train else "_test") if split is None else f"_{split}"
+        self._register_deprecated_attributes()
```

Comment:
Why is this a separate method when it's just used once? Surely it will read better if it's in line. No worries if you disagree,  it's a minor point